.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,233 @@
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
+          playwright install
+
+      - name: Run heartbeat test
+        run: |
+          python actions-hub/scripts/smoke-tests/test_heartbeat.py \
+            --base-url "${{ inputs.target_url }}"
+
+  validate-plugins:
+    if: ${{ inputs.run_test == 'All' ||  inputs.run_test == 'validate-plugins'}}
+    name: Validate eox plugins
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout actions-hub repo
+        uses: actions/checkout@v4
+        with:
+          repository: nelc/actions-hub
+          ref: and/add-basic-smoke-tests
+          path: actions-hub
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
+          playwright install
+
+      - name: Validate eox plugins
+        run: |
+          python actions-hub/scripts/smoke-tests/test_eox_plugins.py \
+            --base-url "${{ inputs.target_url }}" \
+            --plugins ${{ inputs.target_plugins }}
+
+  login-user:
+    if: ${{ inputs.run_test == 'All' ||  inputs.run_test == 'login-user'}}
+    name: Login User Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout actions-hub repo
+        uses: actions/checkout@v4
+        with:
+          repository: nelc/actions-hub
+          ref: and/add-basic-smoke-tests
+          path: actions-hub
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
+          playwright install
+
+      - name: Hide the inputs values to keep them private in the logs when running this workflow
+        uses: levibostian/action-hide-sensitive-inputs@v1
+        with:
+          exclude_inputs: target_url
+
+      - name: Run login-user test
+        run: |
+          python actions-hub/scripts/smoke-tests/test_login_user.py \
+            --base-url "${{ inputs.target_url }}" \
+            --email "${{ inputs.user_email }}" \
+            --password "${{ inputs.user_password }}"
+
+  view-course:
+    if: ${{ inputs.run_test == 'All' ||  inputs.run_test == 'view-course'}}
+    name: View Course Page Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout actions-hub repo
+        uses: actions/checkout@v4
+        with:
+          repository: nelc/actions-hub
+          ref: and/add-basic-smoke-tests
+          path: actions-hub
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
+          playwright install
```

Comment:
Do you think there is a way to not repeat these steps? What are being used in all the jobs.?

Or maybe use some cache for python or something like that, at least to reduce the time of reconfiguring and installing Python and dependencies?

Reply:
Done https://github.com/nelc/actions-hub/pull/12/commits/cad54438c6c2c8cff097025404bf3f8725d9c193